wolfcrypt/src/aes.c

```diff
@@ -12975,6 +12976,29 @@ int wc_AesXtsEncryptUpdate(XtsAes* xaes, byte* out, const byte* in, word32 sz,
     return ret;
 }
 
+int wc_AesXtsEncryptUpdate(XtsAes* xaes, byte* out, const byte* in, word32 sz,
+                           byte *i)
+{
+    if (sz & ((word32)AES_BLOCK_SIZE - 1U))
+        return BAD_FUNC_ARG;
+    return AesXtsEncryptUpdate(xaes, out, in, sz, i);
+}
+
+int wc_AesXtsEncryptFinal(XtsAes* xaes, byte* out, const byte* in, word32 sz,
```

Comment:
What value does "final" add? Is it the zero'ing of "i"?

Reply:
it's a couple things:

(1) lets us add error-checking for all calls to `Update()` to make sure they're correctly block-aligned (trying to get ahead of ZenDesk tickets on that).

(2) the zeroing lets us check for API abuse, in that it guarantees a wrong result if the user calls Update after a Final, and of course it also prevents anything valuable from leaking out through the IV.